util/topgen/merge.py

```diff
@@ -885,20 +889,36 @@ def create_alert_lpgs(top, name_to_block: Dict[str, IpBlock]):
 
         # if clock group is "unique", add some uniquification to the tag
         lpg_name = f"{module['name']}_{lpg_name}" if unique_cg else lpg_name
-        if lpg_name not in lpg_dict:
-            lpg_dict.update({lpg_name: num_lpg})
-            # since the alert handler can tolerate timing delays on LPG
-            # indication signals, we can just use the clock / reset signals
-            # of the first block that belongs to a new unique LPG.
-            clock = module['clock_connections'][block_clock.clock]
-            top['alert_lpgs'].append({
+
+        def append_to_lpg_dict(lpg_dict):
+            lpg_dict.append({
                 'name': lpg_name,
                 'clock_group': None if unmanaged_clock else clock_group,
                 'clock_connection': clock,
                 'unmanaged_clock': unmanaged_clock,
                 'reset_connection': primary_reset
             })
-            num_lpg += 1
+
+        if 'outgoing_alert' in module:
+            alert_group = module['outgoing_alert']
+            if lpg_name not in outgoing_lpg_dict[alert_group]:
+                outgoing_lpg_dict[alert_group].update({lpg_name: outgoing_num_lpg[alert_group]})
+
+                # since the alert handler can tolerate timing delays on LPG
+                # indication signals, we can just use the clock / reset signals
+                # of the first block that belongs to a new unique LPG.
+                clock = module['clock_connections'][block_clock.clock]
```

Comment:
The flow of information through the `clock` variable is pretty confusing. Can we just add it as an argument to `append_to_lpg_dict`?